app/javascript/containers/NewFloofContainer.js

```diff
@@ -1,9 +1,225 @@
-import React from 'react';
+import React, { Component } from 'react'
+import { browserHistory } from 'react-router'
+import FormField from '../components/FormField'
+import CategoryField from '../components/CategoryField'
 
-const NewFloofContainer = (props) => {
+class NewFloofContainer extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      floof: {
+        name: '',
+        job_title: '',
+        current_employer: '',
+        category_id: '',
+        species: ''
+      },
+      message: '',
+      categories: [],
+      errors: []
+    }
+    this.handleNameChange = this.handleNameChange.bind(this)
+    this.handleJobChange = this.handleJobChange.bind(this)
+    this.handleEmployerChange = this.handleEmployerChange.bind(this)
+    this.handleSpeciesChange = this.handleSpeciesChange.bind(this)
+    this.handleCategoryChange = this.handleCategoryChange.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.handleClear = this.handleClear.bind(this)
+    this.validateName = this.validateName.bind(this)
+    this.validateJobTitle = this.validateJobTitle.bind(this)
+    this.validateSpecies = this.validateSpecies.bind(this)
+    this.clearErrors = this.clearErrors.bind(this)
+  }
 
-  return(
-    <h1>Hello from New Floof container</h1>
-  )}
+  componentDidMount() {
+    fetch('/api/v1/categories.json')
+    .then(response => {
+      if (response.ok) {
+        return response
+      } else {
+        let errorMessage = `${response.status} (${response.statusText})`,
+        error = new Error(errorMessage)
+        throw(error)
+      }
+    })
+    .then(response => response.json())
+    .then(body => {
+      this.setState({ categories: body.categories })
+    })
+    .catch(error => console.error(`Error in fetch: ${error.message}`))
+  }
 
-export default NewFloofContainer;
+  handleNameChange(event) {
+    this.setFormDetail('name', event.target.value )
+  }
+  handleJobChange(event) {
+    this.setFormDetail('job_title', event.target.value )
+  }
+  handleEmployerChange(event) {
+    this.setFormDetail('current_employer', event.target.value )
+  }
+  handleSpeciesChange(event) {
+    this.setFormDetail('species', event.target.value )
+  }
+  handleCategoryChange(event) {
+    this.setFormDetail('category_id', event.target.value )
+  }
+
+  clearErrors() {
+    this.setState({
+      errors: []
+    })
+  }
+
+  validateName(){
+    if(this.state.floof.name === '') {
+      return ["Name can't be blank"]
+    }
+    else {
+      return []
+    }
+  }
+
+  validateJobTitle(){
+    if(this.state.floof.job_title === '') {
+      return ["Job Title can't be blank"]
+    }
+    else {
+      return []
+    }
+  }
+
+  validateSpecies(){
+    if(this.state.floof.species === '') {
+      return ["Species can't be blank"]
+    }
+    else {
+      return []
+    }
+  }
```

Comment:
Could we refactor these functions to actually update our state, rather than continuing to have to return an empty array for three methods in a row?